File: Java/problem1395.java
class Solution {
    public int numTeams(int[] rating) {
        int result = 0;
        for (int middle = 1; middle < rating.length - 1; ++middle) {
            int leftLargerNum = 0, rightLargerNum = 0;
            for (int left = 0; left < middle; ++left) {
                if (rating[left] > rating[middle]) {
                    ++leftLargerNum;
                }
            }
            for (int right = middle + 1; right < rating.length; ++right) {
                if (rating[right] > rating[middle]) {
                    ++rightLargerNum;
                }
            }
            result += (middle-leftLargerNum) * rightLargerNum + leftLargerNum * (rating.length-1-middle-rightLargerNum);
        }
        return result;
    }
}

File: python/problem1395.py
class Solution:
    def numTeams(self, rating: List[int]) -> int:
        N = len(rating)
        result = 0
        for middle in range(1, N - 1):
            leftLargerNum, rightLargerNum = 0, 0
            for left in range(middle):
                if rating[left] > rating[middle]:
                    leftLargerNum += 1
            for right in range(middle + 1, N):
                if rating[right] > rating[middle]:
                    rightLargerNum += 1
            result += (middle - leftLargerNum) * rightLargerNum + leftLargerNum * (N - 1 - middle - rightLargerNum)
        return result
